concepts/0094-cross-domain-messaging/README.md

```diff
@@ -0,0 +1,302 @@
+
+#### Terminology
+
+The following terms are used in this RFC with the following meanings:
+
+- Domain - a set of Agents collaborating on behalf of an Identity
+  - It's assumed that the Agents of a Domain were developed by a single vendor and so MAY have implementation-specific mechanisms for tracking extra information about one another.
+  - An example of two Domains is provided in the image below.
+- Edge Agents - the Agents involved in sending (creating) and receiving (processing) an Agent Message
+  - Sender - the Agent sending an original Agent Message
+  - Receiver - the Agent intended to receive an Agent Message from the Sender
+  - **Note**: A message MAY pass through many Agents between the Sender and Receiver
+- Domain Endpoint - a physical endpoint for messages into domains that MUST be assumed to be shared across many Identities
+  - Shared by many identities (e.g. https://endpoint.agentsRus.com)
+  - Agency - the handler for messages sent to the Domain Endpoint.
+- Mediators - agents explicitly known to the sender as part of routing messages to a Receiver.
+- DID - reference to the literal Decentralized ID text
+  - e.g. did:sov:1234abcd
+- DID#keyname - reference to the DID appended with "#" and a specific key from the DIDDoc
+  - e.g. did:sov:1234abcd#1 references key "1" of the "did:sov:1234abcd" DIDDoc.
+  - **Note**: The #keyname is NOT the actual Public Key - it's a reference to an entry in the DIDDoc that contains the Public Key.
+
+#### DIDDoc
+
+The term "DIDDoc" is used in this RFC as it is defined in the [DID Specification](https://w3c-ccg.github.io/did-spec/):
+
+- a collection of public keys and endpoints,
+- controlled by an identity,
+- associated with a DID, and
+- used for a relationship.
+
+A DID can be resolved to get its corresponding DIDDoc by any Agent that needs access to the DIDDoc. This is true whether talking about a DID on a Public Ledger, or a pairwise DID persisted only to the parties of the relationship. In the case of pairwise DIDs, it's the (implementation specific) domain's responsibility to ensure such resolution is available to all Agents requiring it.
+
+#### Messages are Private
+
+Agent Messages sent from a Sender to a Receiver SHOULD be private. That is, the Sender SHOULD encrypt the message with a public key for the Receiver. Any agent in between the Sender and Receiver will know only to whom the message is intended (by DID and possibly keyname within the DID), not anything about the message.
+
+#### The Sender Knows The Receiver
+
+This RFC assumes that the Sender knows the Receiver's DID and, within the DIDDoc for that DID, the keyname to use for the Receiver's Agent. How the Sender knows the DID and keyname to send the message is not defined within this RFC - that is a higher level concern.
+
+The Receiver's DID MAY be a public or pairwise DID, and MAY be on a Public Ledger or a only shared between the parties of the relationship.
```

Comment:
@swcurran "or only shared between the parties of the relationship"

Reply:
Doh - thanks! Fixed.